pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -313,3 +313,79 @@ def test_order_aggregate_multiple_funcs():
     expected = pd.Index(['sum', 'max', 'mean', 'ohlc', 'min'])
 
     tm.assert_index_equal(result, expected)
+
+
+class TestKeywordAggregation:
+
+    def test_agg_relabel(self):
+        df = pd.DataFrame({"group": ['a', 'a', 'b', 'b'],
+                           "A": [0, 1, 2, 3],
+                           "B": [5, 6, 7, 8]})
+        result = df.groupby("group").agg(
+            a_max=("A", "max"),
+            b_max=("B", "max"),
+        )
+        expected = pd.DataFrame({"a_max": [1, 3], "b_max": [6, 8]},
+                                index=pd.Index(['a', 'b'], name='group'),
+                                columns=['a_max', 'b_max'])
+        tm.assert_frame_equal(result, expected)
+
+        # order invariance
+        result = df.groupby('group').agg(
+            b_min=("B", "min"),
+            a_min=("A", min),
+            a_max=("A", "max"),
+            b_max=("B", "max"),
+        )
+        expected = pd.DataFrame({"b_min": [5, 7],
+                                 "a_min": [0, 2],
+                                 "a_max": [1, 3],
+                                 "b_max": [6, 8]},
+                                index=pd.Index(['a', 'b'], name='group'),
+                                columns=['b_min', 'a_min', 'a_max', 'b_max'])
+        if not compat.PY36:
+            expected = expected[['a_max', 'a_min', 'b_max', 'b_min']]
+        tm.assert_frame_equal(result, expected)
+
```

Comment:
In order for this test to cover more ground, could we include a third test with all possible _aggfunc_ types in the test?

Something like:

```python
        #earlier: from functools import partial
        p98 = partial(np.percentile, q=98)
        result = df.groupby('group').agg(
            b_min=("B", "min"),                         # string aggfunc
            a_min=("A", min),                           # built-in
            a_average=("A", np.mean),                   # numpy function
            a_p98=("A", p98),                           # partial function
            b_p98=("B", lambda x: np.percentile(x, 98)) # lambda function
        )
```